### PR TITLE
Use image without -dev suffix for http prober

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

/assign @nikhita

### CODE REVIEW

File: api/pkg/controller/openshift/testdata/Kubermatic API image is overwritten-usercluster-controller.golden.yaml [hunk 1]
```diff
@@ -89,7 +89,7 @@ spec:
         - /bin/cp
         - /usr/local/bin/http-prober
         - /http-prober-bin/http-prober
-        image: quay.io/kubermatic/http-prober:v0.2-dev0
+        image: quay.io/kubermatic/http-prober:v0.2
```
Comment (nikhita) [inline api/pkg/controller/openshift/testdata/Kubermatic API image is overwritten-usercluster-controller.golden.yaml:92]: Can we change this filename to use dashes `-` instead?

**Comment by alvaroaleman** [comment]
@nikhita Updated

File: api/pkg/controller/openshift/openshift_controller_test.go [hunk 2]
```diff
@@ -167,7 +168,7 @@ func TestResources(t *testing.T) {
 				t.Fatalf("failed to serialize object %q: %v", metav1object.GetName(), err)
 			}
 
-			testhelper.CompareOutput(t, fmt.Sprintf("%s-%s", tc.name, metav1object.GetName()), string(serializedObject), *update, ".yaml")
+			testhelper.CompareOutput(t, fmt.Sprintf("%s-%s", strings.Replace(tc.name, " ", "_", -1), metav1object.GetName()), string(serializedObject), *update, ".yaml")
```
Comment (nikhita) [inline api/pkg/controller/openshift/openshift_controller_test.go:171]: Why not use underscores directly in the filename? :thinking:

Comment (alvaroaleman) [reply api/pkg/controller/openshift/openshift_controller_test.go:171]: Its not a filename, its a testcase name and its harder to read with underscores IMO

**Pull request merged**